packages/ui/src/fontManager.ts

```diff
@@ -0,0 +1,174 @@
+import axios from "axios";
+import { getContribLogger } from "@playkit-js-contrib/common";
+import PlayerConfig = KalturaPlayerTypes.PlayerConfig;
+
+const logger = getContribLogger({
+    module: "ui",
+    class: "FontManager"
+});
+
+const FontKeyPrefix: string = "contrib-plugins-font-";
+
+export class FontManager {
+    private static _instance: FontManager;
+
+    private _loadedFontName: string;
+
+    private constructor() {}
+
+    public static getInstance(): FontManager {
+        if (!FontManager._instance) {
+            FontManager._instance = new FontManager();
+        }
+        return FontManager._instance;
+    }
+
+    public loadFont(data: PlayerConfig.Fonts): void {
+        // if download data exists
+        if (data.downloadData && data.downloadData.name && data.downloadData.url) {
```

Comment:
what expected in a scenario where each plugin requests a different font ?

Reply:
it depends how far you want to take it. Each player has a unique identifier. To support multiple players you will need to scope the overrides of the player to a classname unique to each player. Only then you will be able to add smart download checks that make sure you don't download the same font twice.

But to meet our current requirements and time frame, you can just check if the second (third, forth...) is asking for a different font. if he does then you can just add a friendly yet descriptive warning saying that other player won and you are not going to handle his request. Still you must set the css overrides regardless since the fontfamily includes the fallback fonts list.

Reply:
done